propagation.go

```diff
@@ -178,13 +187,30 @@ func extractTraceContext(opaqueCarrier interface{}) (SpanContext, error) {
 		return spanContext, ot.ErrSpanContextNotFound
 	}
 
-	if !spanContext.Suppressed && (spanContext.SpanID == 0 != (spanContext.TraceIDHi == 0 && spanContext.TraceID == 0)) {
-		return spanContext, ot.ErrSpanContextCorrupted
+	// When the context is not suppressed and one of Instana ID headers set.
+	if !spanContext.Suppressed &&
+		(spanContext.SpanID == 0 != (spanContext.TraceIDHi == 0 && spanContext.TraceID == 0)) {
+		sensor.logger.Debug("only one of Instana ID headers present",
+			" SpanID=", spanContext.SpanID,
+			" TraceID=", spanContext.TraceID,
+			" TraceIDHi=", spanContext.TraceIDHi)
+
+		// Check if w3 context was found
+		if spanContext.W3CContext.IsZero() {
+			return spanContext, ot.ErrSpanContextCorrupted
+		}
+
+		// Reset the trace IDs
+		resetSpanContextIDs(&spanContext)
```

Comment:
I'd rather suggest we explicitly return a new `SpanContext` here that has only the W3C trace and not reuse the existing one. I believe this way it'd be easier to follow the logic and also allow us to not to maintain the `resetSpanContextIDs()` in case we add now fields to the context.

Reply:
👍  I will change it